WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -1,11 +1,72 @@
 # frozen_string_literal: true
 class Api::V0::CompetitionsController < Api::V0::ApiController
-  def show
-    competition = Competition.where(id: params[:id], showAtAll: true).first
-    unless competition
-      render json: { error: "Competition with id #{params[:id]} not found" }, status: 404
-      return
+  def index
+    managed_by_user = nil
+    if params[:managed_by_me].present?
+      require_scope!("manage_competitions")
+      managed_by_user = current_api_user
     end
+
+    competitions = Competition.search(params[:q], params: params, managed_by_user: managed_by_user)
+    competitions = competitions.includes(:delegates, :organizers)
+
+    paginate json: competitions
+  end
+
+  def show
+    competition = competition_from_params
     render json: competition
   end
+
+  def wcif
```

Comment:
`show_wcif` seems more consistent to me with `update_wcif` method right next to it. I've made that change, which did not actually affect the API routes.